test/e2e/persistent_volumes-vsphere.go

```diff
@@ -0,0 +1,170 @@
+				namePrefix: "vspherepv-",
+				pvSource: v1.PersistentVolumeSource{
+					VsphereVolume: &v1.VsphereVirtualDiskVolumeSource{
+						VolumePath: volumePath,
+						FSType:     "ext4",
+					},
+				},
+				prebind: nil,
+			}
+		}
+	})
+
+	AfterEach(func() {
+		framework.Logf("AfterEach: Cleaning up test resources")
+		if c != nil {
+			deletePodWithWait(f, c, clientPod)
+			pvPvcCleanup(c, ns, pv, pvc)
+			clientPod = nil
+			pvc = nil
+			pv = nil
+		}
+	})
+
+	AddCleanupAction(func() {
+		if len(volumePath) > 0 {
+			vsp.DeleteVolume(volumePath)
+		}
+	})
+
+	/*
+		Delete the PVC and then the pod.  Expect the pod to succeed in unmounting and detaching PD on delete.
+
+		Test Steps:
+		1. Create volume (vmdk)
+		2. Create PV with volume path for the vmdk.
+		3. Create PVC to bind with PV.
+		4. Create a POD using the PVC.
+		5. Delete PVC.
+		6. Verify Disk is attached to the node.
+		7. Delete POD, POD deletion should succeed.
+		8. Wait and Verify Disk is Detached from the node.
+
+		Clean up.
+		1. Delete PV
+		2. Delete Volume (vmdk)
+	*/
+
+	It("should test that deleting a PVC before the pod does not cause pod deletion to fail on PD detach", func() {
+		By("Creating the PV and PVC")
+		pv, pvc = createPVPVC(c, pvConfig, ns, false)
+		waitOnPVandPVC(c, ns, pv, pvc)
+
+		By("Creating the Client Pod")
+		clientPod = createClientPod(c, ns, pvc)
+		node := types.NodeName(clientPod.Spec.NodeName)
+
+		By("Verify disk should be attached to the node")
+		isAttached, err := verifyVSphereDiskAttached(vsp, volumePath, node)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(isAttached).To(BeTrue(), "disk is not attached with the node")
+
+		By("Deleting the Claim")
+		deletePersistentVolumeClaim(c, pvc.Name, ns)
+
+		By("Deleting the Pod")
+		deletePodWithWait(f, c, clientPod)
+
+		By("Verifying Persistent disk is detached")
+		waitForVSphereDiskToDetach(vsp, volumePath, node)
+	})
+
+	/*
+		Delete the PV and then the pod.  Expect the pod to succeed in unmounting and detaching PD on delete.
+
+		Test Steps:
+		1. Create volume (vmdk)
+		2. Create PV with volume path for the vmdk.
+		3. Create PVC to bind with PV.
+		4. Create a POD using the PVC.
+		5. Delete PV.
+		6. Verify Disk is attached to the node.
+		7. Delete POD, POD deletion should succeed.
+		8. Wait and Verify Disk is Detached from the node.
+
+		Clean up.
+		1. Delete PVC
+		2. Delete Volume (vmdk)
+	*/
+	It("should test that deleting the PV before the pod does not cause pod deletion to fail on PD detach", func() {
+		By("Creating the PV and PVC")
+		pv, pvc = createPVPVC(c, pvConfig, ns, false)
+		waitOnPVandPVC(c, ns, pv, pvc)
+
+		By("Creating the Client Pod")
+		clientPod = createClientPod(c, ns, pvc)
+		node := types.NodeName(clientPod.Spec.NodeName)
+
+		By("Verify disk should be attached to the node")
+		isAttached, err := verifyVSphereDiskAttached(vsp, volumePath, node)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(isAttached).To(BeTrue(), "disk is not attached with the node")
```

Comment:
L148-159 are identical to the above `It` block's beginning lines. Can this code be moved to a common func?